File: packages/python/examples/simple_agent.py
"""Basic EchoProvider agent example."""

from handoffkit import Agent


def main() -> None:
    """Run a deterministic local agent with the built-in EchoProvider."""
    agent = Agent(
        name="Planner",
        role="Create concise implementation plans with decisions and next steps.",
    )
    result = agent.run("Create a plan for a Python CLI app with tests.")
    print(result)


if __name__ == "__main__":
    main()

File: js/packages/core/examples/simple_agent.js
/**
 * Basic EchoProvider agent example in JavaScript.
 */
import { Agent } from "../src/index.js";

function main() {
  const agent = new Agent({
    name: "Planner",
    role: "Create concise implementation plans with decisions and next steps.",
  });

  const result = agent.run("Create a plan for a JS app with tests.");
  console.log("Agent Name:", result.agentName);
  console.log("Task:", result.task);
  console.log("Final Output:\n", result.finalOutput);
}

main();
